streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -544,11 +306,7 @@ public static StreamThread create(final InternalTopologyBuilder builder,
 
         final ThreadCache cache = new ThreadCache(logContext, cacheSizeBytes, streamsMetrics);
 
-        final Map<TaskId, Producer<byte[], byte[]>> taskProducers = new HashMap<>();
```

Comment:
Managed fully inside the task factory now.

You'll also notice a bunch of references to the producers are similarly gone in the following lines.